gitCoreJGit/src/main/java/com/virtuslab/gitcore/impl/jgit/GitCoreBranch.java

```diff
@@ -37,6 +38,8 @@
   protected final GitCoreRepository repo;
   protected final String branchName;
 
+  private final AtomicReference<GitCoreCommit> pointedCommitRef = new AtomicReference<>(null);
```

Comment:
...and `@Nullable` before `GitCoreCommit` + maybe mark `AtomicReference#get` as `@Nullable` in Checker stubs